app/src/main/java/com/mybus/activity/MainActivity.java

```diff
@@ -751,6 +754,12 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
                         String busLineName = data.getStringExtra(DisplayBusLinesActivity.RESULT_BUS_LINE_NAME);
                         showCompleteBusRoute(busLineId, busLineName);
                         break;
+                    case DISPLAY_BUS_LINES_RESULT:
+                        int busResultId = data.getIntExtra(BusResultsActivity.SELECTED_BUS_LINE_EXTRA, -1);
```

Comment:
Also return the new results list (in case of new searches). This is necessary to remove the mBusResults var.